File: src/main/java/com/loadbalancing/kiosk/domain/qna/infra/entity/Qna.java
package com.loadbalancing.kiosk.domain.qna.infra.entity;

import com.loadbalancing.kiosk.global.entity.BaseSoftDeleteTimeEntity;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

@Getter
@Entity
@Builder
@Table(name = "qna")
@SQLDelete(sql = "UPDATE qna SET deleted_at = CURRENT_TIMESTAMP WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class Qna extends BaseSoftDeleteTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(columnDefinition = "TEXT")
    private String answer;

    @Column(nullable = false)
    private boolean answered;

    public void answer(String answer) {
        this.answer = answer;
        this.answered = true;
    }

}

File: client/types/qna.ts
// QnaRequest.QnaCreateRequest (백엔드)
export interface QnaCreateRequest {
  email: string;
  title: string;
  content: string;
}

// QnaRequest.QnaAnswerRequest (백엔드)
export interface QnaAnswerRequest {
  answer: string;
}

// QnaResponse.QnaInfo (백엔드) - 목록 조회에 email 필터가 없어서(GET /api/v1/auth/qna),
// "내 문의만" 보는 게 아니라 게시판처럼 전체 문의가 다 보이는 구조
export interface QnaInfo {
  id: number;
  email: string;
  title: string;
  content: string;
  answer: string | null;
  answered: boolean;
  createdAt: string;
}
